### PR TITLE
feat: wallet ffi use dns

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Add the option for the wallet_ffi, to use the DNS build into it. 

Motivation and Context
---
Allow the fii to use DNS. 

How Has This Been Tested?
---
Manual

### CODE REVIEW

File: base_layer/wallet_ffi/src/lib.rs [hunk 1]
```diff
@@ -6365,6 +6379,45 @@ pub unsafe extern "C" fn wallet_set_base_node_peer(
...
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// `TariPublicKeys` - Returns a list of all known public keys
+///
+/// # Safety
+/// None
+#[no_mangle]
+pub unsafe extern "C" fn wallet_get_seed_peers(wallet: *mut TariWallet, error_out: *mut c_int) -> *mut TariPublicKeys {
```
Comment (brianp) [inline base_layer/wallet_ffi/src/lib.rs:6391]: What's the purpose of the function? So they can be listed as connectable base nodes?

File: base_layer/wallet_ffi/src/lib.rs [hunk 2]
```diff
@@ -5259,6 +5259,7 @@ pub unsafe extern "C" fn wallet_create(
     passphrase: *const c_char,
     seed_words: *const TariSeedWords,
     network_str: *const c_char,
+    peer_seed_str: *const c_char,
```
Comment (brianp) [inline base_layer/wallet_ffi/src/lib.rs:5262]: I don't think we need a whole parameter for this. It's based on the network, and we already have that:

```
dns_seeds: StringList::from(vec![format!("seeds.{}.tari.com", network.as_key_str())])
```

File: base_layer/wallet_ffi/src/lib.rs [hunk 3]
```diff
@@ -5473,6 +5486,7 @@ pub unsafe extern "C" fn wallet_create(
     let peer_seeds = PeerSeedsConfig {
         dns_seeds_name_server: DEFAULT_DNS_NAME_SERVER.parse().unwrap(),
         dns_seeds_use_dnssec: true,
```
Comment (brianp) [inline base_layer/wallet_ffi/src/lib.rs]: From my chat based experience- dnssec needs to be turned off or all our DNS validations will fail. 

This is the chat config for PeerSeeds which *does* use DNS (including the existing integrated wallet branch)

```
        peer_seeds: PeerSeedsConfig {
            dns_seeds_use_dnssec: false,
            dns_seeds_name_server: DEFAULT_DNS_NAME_SERVER.parse().unwrap(),
            dns_seeds: StringList::from(vec![format!("seeds.{}.tari.com", network.as_key_str())]),
            ..PeerSeedsConfig::default()
        },
        ```

**Review by brianp — DISMISSED** [review]
Miner change could reduce overall changes and interface if we interpolate the DNS address based on the network instead of passing it in.

Also a note about dnssec, but if it's been tested #yolo.

File: base_layer/wallet_ffi/src/lib.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/wallet_ffi/src/lib.rs:6391]: yeah

File: base_layer/wallet_ffi/src/lib.rs (hunk 2 above)
Comment (SWvheerden) [reply base_layer/wallet_ffi/src/lib.rs:5262]: thats only if we dont change it...

File: base_layer/wallet_ffi/src/lib.rs (hunk 3 above)
Comment (SWvheerden) [reply base_layer/wallet_ffi/src/lib.rs]: yip mapped that through

**Review by brianp — APPROVED** [review]

**Pull request merged**